Various cleanups

fingolfin opened the pull request:
These changes were created as part of the hpc-merge project. But they make sense on their own, and it will simplify my life a bit to have them inside of the master branch, so I am submitting thm as a PR now.

ChrisJefferson commented:
This all looks fine to me. This might break some stupid non-POSIX compatible system, but we can fix that if and when they come forward.

fingolfin commented:
I just added two more commits, and also fixed two older ones where I forgot to run `cd cnf && make`. Sorry for doing that post review, but it should be stable from here on.
